packages/next-auth/src/core/routes/session.ts

```diff
@@ -123,19 +130,16 @@ export default async function session(
         }
 
         // Pass Session through to the session callback
-        // @ts-expect-error
         const sessionPayload = await callbacks.session({
           // By default, only exposes a limited subset of information to the client
           // as needed for presentation purposes (e.g. "you are logged in as...").
           session: {
-            user: {
-              name: user.name,
-              email: user.email,
-              image: user.image,
-            },
+            user: { name: user.name, email: user.email, image: user.image },
             expires: session.expires.toISOString(),
           },
           user,
+          trigger: isUpdate ? "update" : "database",
```

Comment:
why do we need `trigger: "database"` or `trigger: "jwt"` ? As far as I concern, if the developers set the strategy, they already know about this info, so it's kinda redundant IMO

Reply:
For discriminated union types. trying to make TS happy. Our current typing is not safe as I've seen people trying it use either `user` or `token` when it's not available because of the strategy they were using.

Reply:
Addressed in https://github.com/nextauthjs/next-auth/pull/7056/commits/3c9b49411c72a2afa05c97763cdf4fbbe434c74e

Maybe it's not worth breaking current types for this. We can do it in `@auth/core` instead.